urllib3/response.py

```diff
@@ -182,6 +195,53 @@ def tell(self):
         """
         return self._fp_bytes_read
 
+    def _init_length(self, request_method):
+        """
+        Set initial length value for Response content.
+
+        Logic extracted from httplib.py in the standard Python library.
+        """
+        length = self.headers.get('content-length')
+
+        if length is not None and self.chunked:
+            # This Response will fail with an IncompleteRead if it can't be
+            # received as chunked. This works as a fallback to at least attempt
+            # to read the response before raising an exception.
+            log.warning("Received response with both Content-Length and "
+                        "Transfer-Encoding set. This is expressly forbidden "
+                        "by RFC 7230 sec 3.3.2. Ignoring Content-Length and "
+                        "attempting to process response as Transfer-Encoding: "
+                        "chunked.")
+
+        if length is not None and not self.chunked:
+            try:
+                lengths = set([int(float(val)) for val in length.split(',')])
+                if len(lengths) > 1:
+                    raise InvalidHeader("Content-Length contained multiple, "
+                                        "unmatching values (%s)" % length)
+                length = lengths.pop()
```

Comment:
`set` doesn't support indexing but we know it's a single value at this point. We could do a dictionary too, I just figured a `set` made more sense since we're not utilizing the value portion of `dict`.